### PR TITLE
Migrate credential retrieval code and demos to match spec

### PR DESCRIPTION
**Pull request by iainmcgin**
Stacked on top of #30.

### CODE REVIEW

File: api/java/org/openyolo/api/CredentialClient.java [hunk 1]
```diff
@@ -190,40 +192,76 @@ private Intent createSaveIntent(
...
-            return null;
+            return Credential.fromProtoBytes(resultBytes);
+        } catch (IOException ex) {
+            Log.e(LOG_TAG, "hint result is malformed, returning default response");
+            return createDefaultHintRetrieveResult();
         }
     }
 
+    @NonNull
+    private CredentialRetrieveResult createDefaultCredentialRetrieveResult() {
```
Comment (dxslly) [inline api/java/org/openyolo/api/CredentialClient.java]: Nit: Can be static.

File: api/java/org/openyolo/api/CredentialClient.java [hunk 2]
```diff
@@ -190,40 +192,76 @@ private Intent createSaveIntent(
...
+        return new CredentialRetrieveResult.Builder(CredentialRetrieveResult.RESULT_UNKNOWN)
+                .build();
+    }
+
+    @Nullable
+    private Credential createDefaultHintRetrieveResult() {
```
Comment (dxslly) [inline api/java/org/openyolo/api/CredentialClient.java]: Nit: Can be static.

File: api/javatests/org/openyolo/api/CredentialClientTest.java [hunk 3]
```diff
@@ -295,31 +293,34 @@ public void getSaveIntent_noKnownProviders() throws Exception {
     }
 
     @Test
-    public void getCredentialFromActivityResult_noExtras() throws Exception {
+    public void testGetCredentialRetrieveResult() throws Exception {
```
Comment (dxslly) [inline api/javatests/org/openyolo/api/CredentialClientTest.java]: Please remove test prefix.  It's not standardized in this codebase, but I'd really like to follow a naming pattern for unit tests moving forward. <method_name>_<input_qualifiers>_<expected_result> is the template I use.  What do you think about adopting it?

File: api/javatests/org/openyolo/api/CredentialClientTest.java [hunk 4]
```diff
@@ -295,31 +293,34 @@ public void getSaveIntent_noKnownProviders() throws Exception {
...
+        assertThat(result.getCredential()).isNotNull();
+        assertThat(result.getCredential().getIdentifier())
+                .isEqualTo(testCredential.getIdentifier());
+
+        assertThat(result.getAdditionalProps()).isEmpty();
     }
 
     @Test
-    public void getCredentialFromActivityResult_withExtras() throws Exception {
+    public void testGetCredentialRetrieveResult_noExtra() throws Exception {
```
Comment (dxslly) [inline api/javatests/org/openyolo/api/CredentialClientTest.java]: Similarly here

File: bbq/proto/bbq.proto [hunk 5]
```diff
@@ -52,5 +52,15 @@ message BroadcastQueryResponse {
 }
 
 message ClientVersion {
-    string version = 1;
-}
+    // required
```
Comment (dxslly) [inline bbq/proto/bbq.proto]: Skipped tag id 2

File: protocol/java/org/openyolo/protocol/CredentialRetrieveResult.java [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+            Protobufs.CredentialRetrieveResult.ResultCode.SUCCESS_VALUE;
+
+    /**
+     * Indicates that the credential request was denied by the provider.
+     */
+    public static final int RESULT_REJECTED_BY_PROVIDER =
+            Protobufs.CredentialRetrieveResult.ResultCode.REJECTED_BY_PROVIDER_VALUE;
+
+    /**
+     * Indicates that the credential request was defined by the user.
```
Comment (dxslly) [inline protocol/java/org/openyolo/protocol/CredentialRetrieveResult.java]: s/defined/denied

File: protocol/java/org/openyolo/protocol/CredentialRetrieveResult.java [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+        public Builder setCredential(@Nullable Credential credential) {
+            mCredential = credential;
+            return this;
+        }
+
+        /**
+         * Specifies the credential being returned as part of this result, in protocol buffer
+         * form. Can be null.
+         */
+        public Builder setCredentialFromProto(@Nullable Protobufs.Credential credential) {
```
Comment (dxslly) [inline protocol/java/org/openyolo/protocol/CredentialRetrieveResult.java]: Document throws IOException

File: spi/java/org/openyolo/spi/BaseCredentialQueryReceiver.java [hunk 8]
```diff
@@ -44,19 +44,19 @@ public BaseCredentialQueryReceiver(String logTag) {
 
     @Override
     protected void processQuery(@NonNull Context context, @NonNull BroadcastQuery query) {
-        CredentialRetrieveRequest requestProto;
+        org.openyolo.protocol.Protobufs.CredentialRetrieveRequest requestProto;
```
Comment (dxslly) [inline spi/java/org/openyolo/spi/BaseCredentialQueryReceiver.java]: Nit: can be removed :)

File: api/javatests/org/openyolo/api/CredentialClientTest.java (hunk 3 above)
Comment (iainmcgin) [inline api/javatests/org/openyolo/api/CredentialClientTest.java]: Old habits die hard; the test prefix was JUnit 3 and even though we now use the Test annotation, I still reflexively write the test prefix. I can remove in a follow-up that removes this from all tests.

I like your template, we can move to that.

File: api/java/org/openyolo/api/CredentialClient.java (hunk 1 above)
Comment (iainmcgin) [inline api/java/org/openyolo/api/CredentialClient.java]: Done

File: api/java/org/openyolo/api/CredentialClient.java (hunk 2 above)
Comment (iainmcgin) [inline api/java/org/openyolo/api/CredentialClient.java]: Done

File: api/javatests/org/openyolo/api/CredentialClientTest.java (hunk 4 above)
Comment (iainmcgin) [inline api/javatests/org/openyolo/api/CredentialClientTest.java]: Ack

File: bbq/proto/bbq.proto (hunk 5 above)
Comment (iainmcgin) [inline bbq/proto/bbq.proto]: Done

File: spi/java/org/openyolo/spi/BaseCredentialQueryReceiver.java (hunk 8 above)
Comment (iainmcgin) [inline spi/java/org/openyolo/spi/BaseCredentialQueryReceiver.java]: *shakes fist at refactoring tools*

**Pull request merged**